OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2226,6 +2231,42 @@ void testAutomaticJointReversal()
     ASSERT_EQUAL(accErr, 0.0, sqrt(integ_accuracy));
 }
 
+void testUserJointReversal()
+{
+    using namespace OpenSim;
+
+    cout << "\n==========================================================="
+         << "\n Test joint reversal set by user in old model file"
+         << "\n==========================================================="
+         << endl;
+
+    // Open model.
+    auto model = Model("double_pendulum_testReverse.osim");
+    //model.finalizeFromProperties();
+    model.initSystem();
+
+    // The topology is specified as follows:
+    //     [ground] <- (pin1) <- [rod1] <- (pin2) <- [rod2]
+    // but the "reverse" element of "pin2" is "true" so, after deserialization,
+    // the topology should be the following:
+    //     [ground] <- (pin1) <- [rod1] -> (pin2) -> [rod2]
```

Comment:
I was just making sure I understood the test. I was interpreting the *topology* according to the `MultibodySystem` which isn't strictly valid, but from the user perspective your comment is fine and the test to verify the correct sense of the Joint is sufficient. Thanks!